### DISCUSSION TITLE
How to use beaMouseDown to provide visual feedback as to selected beat

### DISCUSSION THREAD

**Discussion by garthholmes** [CoderLine/alphaTab#1767] [Q&A] [answered]
Hi,

I have AlphaTab setup and operational in my website and it is working well, with the latest beta 1.3.1, however I would like to provide visual feedback as to what beat has currently been selected, in that I would like to have the same colour applied to a beat, as if it was being played as a song, since beats change colour when clicked but only whilst a song is currently playing.

The code I have found is as follows:

`api.beatMouseDown.on((beat) => {
    startSelectionOnBeat(beat);
});`

from the following page:

https://alphatab.net/docs/reference/api/beatmousedown/

but I get this error:

"Uncaught ReferenceError: startSelectionOnBeat is not defined".

is there somewhere in particular that this code must be used, since I thought if it started with .api it could be placed anywhere after const api= is defined.

Regards

**Comment by Danielku15** [comment, marked as answer]
Unfortunately what you ask for, is not supported. At this point is no API or feature to color individual elements manually.You can follow the discussion over at https://github.com/CoderLine/alphaTab/discussions/1725 to find out more about the plans to support custom coloring.
